--- core/src/main/java/com/electronwill/nightconfig/core/file/AutosaveFileConfig.java	2018-06-26 15:23:51.570841000 +0200
+++ android/core_android/src/main/java/com/electronwill/nightconfig/core/file/AutosaveFileConfig.java	2018-09-24 19:13:33.137133000 +0200
@@ -4,7 +4,6 @@
 import com.electronwill.nightconfig.core.utils.ObservedMap;
 
 import java.io.File;
-import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
 
@@ -48,11 +47,6 @@
 	}
 
 	@Override
-	public Path getNioPath() {
-		return config.getNioPath();
-	}
-
-	@Override
 	public void save() {
 		config.save();
 	}
