--- core/src/main/java/com/electronwill/nightconfig/core/file/FileNotFoundAction.java
+++ android/core_android/src/main/java/com/electronwill/nightconfig/core/file/FileNotFoundAction.java
@@ -1,15 +1,10 @@
 package com.electronwill.nightconfig.core.file;
 
 import com.electronwill.nightconfig.core.ConfigFormat;
+import com.electronwill.nightconfig.core.io.Files;
 
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.IOException;
-import java.io.InputStream;
+import java.io.*;
 import java.net.URL;
-import java.nio.file.Files;
-import java.nio.file.NoSuchFileException;
-import java.nio.file.Path;
 
 /**
  * Defines the action to perform when the file is not found.
@@ -26,7 +21,7 @@
 	 *
 	 * @throws IOException if an IO error occurs
 	 */
-	boolean run(Path file, ConfigFormat<?> configFormat) throws IOException;
+	boolean run(File file, ConfigFormat<?> configFormat) throws IOException;
 
 	// --- Static members ---
 
@@ -37,7 +32,7 @@
 	};
 	FileNotFoundAction READ_NOTHING = (f,c) -> false;
 	FileNotFoundAction THROW_ERROR = (f,c) -> {
-		throw new NoSuchFileException(f.toAbsolutePath().toString());
+		throw new FileNotFoundException(f.toString());
 	};
 
 	/**
@@ -60,21 +55,6 @@
 	 * @return a FileNotFoundAction that copies the file's data if the file is not found
 	 */
 	static FileNotFoundAction copyData(File file) {
-		// copyData(new FIS(file)) isn't used here to avoid dealing with the exception
-		// declared by the FIS constructor
-		return (f,c) -> {
-			Files.copy(new FileInputStream(file), f);
-			return true;
-		};
-	}
-
-	/**
-	 * Action: copies the specified file.
-	 *
-	 * @param file the data url
-	 * @return a FileNotFoundAction that copies the file's data if the file is not found
-	 */
-	static FileNotFoundAction copyData(Path file) {
 		return (f,c) -> {
 			Files.copy(file, f);
 			return true;
