--- core/src/main/java/com/electronwill/nightconfig/core/ConfigFormat.java	2019-04-17 23:33:19.122643757 +0200
+++ android/core_android/src/main/java/com/electronwill/nightconfig/core/ConfigFormat.java	2019-04-17 23:36:00.753849017 +0200
@@ -3,13 +3,12 @@
 import com.electronwill.nightconfig.core.io.ConfigParser;
 import com.electronwill.nightconfig.core.io.ConfigWriter;
 import com.electronwill.nightconfig.core.utils.WriterSupplier;
+import com.electronwill.nightconfig.core.utils.Supplier;
 
 import java.io.File;
+import java.io.FileWriter;
 import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Map;
-import java.util.function.Supplier;
 
 /**
  * A configuration format, that can parse, create and write some types of configurations.
@@ -87,18 +86,8 @@
 	 *
 	 * @param f the existing file to initialize
 	 */
-	default void initEmptyFile(Path f) throws IOException {
-		initEmptyFile(() -> Files.newBufferedWriter(f));
-	}
-
-	/**
-	 * Initializes an empty configuration file so that it can be parsed to an empty configuration.
-	 * Does nothing by default.
-	 *
-	 * @param f the existing file to initialize
-	 */
 	default void initEmptyFile(File f) throws IOException {
-		initEmptyFile(f.toPath());
+		initEmptyFile(() -> new FileWriter(f));
 	}
 
 	/**
