--- core/src/main/java/com/electronwill/nightconfig/core/UnmodifiableConfig.java
+++ android/core_android/src/main/java/com/electronwill/nightconfig/core/UnmodifiableConfig.java
@@ -1,9 +1,10 @@
 package com.electronwill.nightconfig.core;
 
+import com.electronwill.nightconfig.core.utils.Supplier;
+import com.electronwill.nightconfig.core.utils.IntSupplier;
+import com.electronwill.nightconfig.core.utils.LongSupplier;
+
 import java.util.*;
-import java.util.function.IntSupplier;
-import java.util.function.LongSupplier;
-import java.util.function.Supplier;
 
 import static com.electronwill.nightconfig.core.NullObject.NULL_OBJECT;
 import static com.electronwill.nightconfig.core.utils.StringUtils.split;
