--- core/src/main/java/com/electronwill/nightconfig/core/conversion/ConvertedFormat.java	2019-04-17 23:33:19.122643757 +0200
+++ android/core_android/src/main/java/com/electronwill/nightconfig/core/conversion/ConvertedFormat.java	2019-04-17 23:50:18.627399028 +0200
@@ -4,10 +4,10 @@
 import com.electronwill.nightconfig.core.ConfigFormat;
 import com.electronwill.nightconfig.core.io.ConfigParser;
 import com.electronwill.nightconfig.core.io.ConfigWriter;
+import com.electronwill.nightconfig.core.utils.Supplier;
 
 import java.util.Map;
 import java.util.function.Predicate;
-import java.util.function.Supplier;
 
 /**
  * @author TheElectronWill
