--- core/src/main/java/com/electronwill/nightconfig/core/conversion/ObjectConverter.java	2018-11-18 13:34:45.519016000 +0100
+++ android/core_android/src/main/java/com/electronwill/nightconfig/core/conversion/ObjectConverter.java	2018-11-18 13:38:24.022338000 +0100
@@ -3,10 +3,10 @@
 import com.electronwill.nightconfig.core.Config;
 import com.electronwill.nightconfig.core.ConfigFormat;
 import com.electronwill.nightconfig.core.UnmodifiableConfig;
+import com.electronwill.nightconfig.core.utils.Supplier;
 
 import java.lang.reflect.*;
 import java.util.*;
-import java.util.function.Supplier;
