--- core/src/main/java/com/electronwill/nightconfig/core/conversion/ObjectConverter.java
+++ android/core_android/src/main/java/com/electronwill/nightconfig/core/conversion/ObjectConverter.java
@@ -3,10 +3,10 @@
 import com.electronwill.nightconfig.core.Config;
 import com.electronwill.nightconfig.core.ConfigFormat;
 import com.electronwill.nightconfig.core.UnmodifiableConfig;
+import com.electronwill.nightconfig.core.utils.Supplier;
 
 import java.lang.reflect.*;
 import java.util.*;
-import java.util.function.Supplier;
 
 /**
  * Converts Java objects to configs and vice-versa.
